experimental/packages/exporter-trace-otlp-grpc/test/OTLPTraceExporter.test.ts

```diff
@@ -331,28 +331,27 @@ describe('when configuring via environment', () => {
     envSource.OTEL_EXPORTER_OTLP_ENDPOINT = '';
     envSource.OTEL_EXPORTER_OTLP_TRACES_ENDPOINT = '';
   });
+  it('should have user-agent header', () => {
```

Comment:
Since the user agent is not intended to be overridden and should exist regardless of whether specific headers are set, is it worth adding an extra test for this? Maybe even adding an extra piece into the "should override global headers config" test below, something like
```
    envSource.OTEL_EXPORTER_OTLP_TRACES_HEADERS =
      'foo=boo,bar=loo,User-Agent=friend';
...
    assert.deepStrictEqual(collectorExporter.metadata?.get('User-Agent'), [
      `OTel-OTLP-Exporter-JavaScript/${VERSION}`,
    ]);
```

Whether that's with the existing test or a new case, it will also show that the user agent exists on the individual exporters, which is part of what you refactored here (previously it didn't work that way)